### ISSUE TITLE
Implement combined exposure and energy dispersion

### ISSUE THREAD

**Issue by cdeil** [gammapy/gammapy#1538]
In #1503 we added energy dispersion to the map evaluator and 3D analysis. It's optional, if an `edisp` is given it's applied (should be more correct, but much slower),  if it's not given it's not applied (should be a bit incorrect, but fast).

There is an approximate way to apply `edisp` not on each evaluation, but to combine it with `exposure` assuming some spectral shape (e.g. power-law with index 2) only once, and then use that "exposure as a function of reco energy" during the analysis. This is a middle ground that's fast, and more correct than completely ignoring exposure. It's commonly done in HESS when using "effective area as a function of reco energy", and I think it is also sometimes used for Fermi-LAT.

As far as I know, doing this isn't supported in Gammapy at all at the moment, neither for 1D nor for 3D analysis. @joleroi - correct for 1D?

We should probably add that option for both 1D and 3D. This is a reminder issue for the v0.9 milestone.

@registerrier added a `make_map_exposure_reco_energy` function with a docstring, but not implemented already:
https://github.com/gammapy/gammapy/blob/7720a7833ec60a04b1eebec6a179be962dbcf483/gammapy/cube/new.py#L128
@registerrier - I'm deleting this now, because probably this is not the API we want, naively I would think that this takes an exposure cube with a true energy axis as input and applies a correction to it. It also needs some thought how the two cases are handled in the evaluator.
The code will always be in the repo history for copy & paste once someone takes on the task of implementing this feature.

**Comment by adonath** [comment]
This would be nice to have, but I think it's no priority at all, not even for v1.0. So I assigned the `wishlist` milestone.

**Comment by cdeil** [comment]
A new version of the Fermi tools was just released, with this new way to treat energy dispersion highlighted as main new feature:
https://fermi.gsfc.nasa.gov/ssc/data/analysis/documentation/Pass8_edisp_usage.html

I didn't read it yet, but @adonath @registerrier and anyone interested - I think we should re-consider if work is needed on energy dispersion or not, so I'm putting this issue on v0.15. If the conclusion is that what we have now concerning energy dispersion is good enough in terms of accuracy and speed, OK, we can move this to next year.

As far as I know, we never had any documented study showing if our energy dispersion works for 1D or 3D analysis. I think the best way to test it would be to simulate a spectral line near the energy threshold where there is energy bias and the energy dispersion is poor with ctools, and then fit that line with Gammapy and check if it comes out correct or not. My guess is that there will be bugs, simply because that's usually the case for something that's complex and hasn't been tested.

**Comment by facero** [comment]
For large edisp matrices, the np.dot product can be long as it is repeated many times in 3D analysis (Nx * Ny pixels). I tried a 3D analysis with 240x240 pixels and an Edisp with 1000 bins in energy and it was very slow (~1 mn for 1 Npred call).

A good test to see if the implementation of Edisp is correct and test perf could be to simulate a gamma-ray line + power-law.
For futur reference there is long discussion here on stackoverflow:
https://stackoverflow.com/questions/42534970/numpy-matrix-product-sparse-matrices
